calculator/src/App.js

```diff
@@ -0,0 +1,173 @@
+import React from "react";
+import "./App.css";
+
+class App extends React.Component {
+  state = { input: "", previousNum: "", currentNum: "", operator: "" };
+
+  handleClick = (e) => {
+    this.setState({
+      input: this.state.input + e.target.value,
+    });
+  };
+  // clears input when clicked
+  handleClear = (e) => {
+    this.setState({ input: "" });
+  };
+
+  handlePlusNegative = (e) => {
```

Comment:
Best practice: if updating the state based on the current state pass a callback function to setState